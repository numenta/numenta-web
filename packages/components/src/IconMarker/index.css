/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.iconMarker {
  /* */
}

.icon {
  composes: mr2 from 'tachyons';
  composes: relative from 'tachyons';
  color: #0277bd;
  top: -0.066rem;
}
