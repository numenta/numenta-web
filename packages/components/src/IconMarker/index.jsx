// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import IconArrowRight from 'react-icons/lib/fa/arrow-circle-o-right'
import React from 'react'

import styles from './index.css'


const IconMarker = ({children, icon}) => (
  <span className={styles.iconMarker}>
    <span className={styles.icon}>
      {icon}
    </span>
    {children}
  </span>
)

IconMarker.propTypes = {
  children: React.PropTypes.node.isRequired,
  icon: React.PropTypes.element,
}

IconMarker.defaultProps = {
  icon: (<IconArrowRight />),
}

export default IconMarker
