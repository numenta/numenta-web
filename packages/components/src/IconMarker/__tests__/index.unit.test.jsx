import IconArrowLeft from 'react-icons/lib/fa/arrow-circle-o-left'
import React from 'react'
import renderer from 'react-test-renderer'

import IconMarker from '../../IconMarker'


describe('IconMarker React component', () => {

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <IconMarker>
        Hello Default
      </IconMarker>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Custom Icon correctly', () => {
    const component = renderer.create(
      <IconMarker icon={<IconArrowLeft />}>
        Hello Icon
      </IconMarker>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
