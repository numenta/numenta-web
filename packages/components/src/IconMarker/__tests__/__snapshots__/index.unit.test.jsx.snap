exports[`IconMarker React component Renders Custom Icon correctly 1`] = `
<span
  className="iconMarker">
  <span
    className="icon">
    <svg
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width="1em">
      <g>
        <path
          d="m28.7 17.9v4.2q0 0.3-0.2 0.5t-0.5 0.3h-7.9v4.2q0 0.4-0.2 0.6t-0.5 0.2q-0.2 0-0.5-0.3l-7.1-7.1q-0.2-0.2-0.2-0.5t0.2-0.5l7.1-7.2q0.2-0.2 0.5-0.2 0.3 0 0.5 0.3t0.2 0.5v4.2h7.9q0.3 0 0.5 0.3t0.2 0.5z m3.6 2.1q0-3.3-1.6-6.1t-4.5-4.4-6.1-1.6-6.1 1.6-4.4 4.4-1.6 6.1 1.6 6.1 4.4 4.4 6.1 1.6 6.1-1.6 4.5-4.4 1.6-6.1z m5 0q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z" />
      </g>
    </svg>
  </span>
  Hello Icon
</span>
`;

exports[`IconMarker React component Renders Default correctly 1`] = `
<span
  className="iconMarker">
  <span
    className="icon">
    <svg
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width="1em">
      <g>
        <path
          d="m28.7 20q0 0.3-0.2 0.5l-7.1 7.2q-0.2 0.2-0.5 0.2-0.3 0-0.5-0.3t-0.3-0.5v-4.2h-7.8q-0.3 0-0.5-0.3t-0.2-0.5v-4.2q0-0.3 0.2-0.5t0.5-0.3h7.8v-4.2q0-0.4 0.2-0.6t0.6-0.2q0.2 0 0.5 0.3l7.1 7.1q0.2 0.2 0.2 0.5z m3.6 0q0-3.3-1.6-6.1t-4.5-4.4-6.1-1.6-6.1 1.6-4.4 4.4-1.6 6.1 1.6 6.1 4.4 4.4 6.1 1.6 6.1-1.6 4.5-4.4 1.6-6.1z m5 0q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z" />
      </g>
    </svg>
  </span>
  Hello Default
</span>
`;
